ext4: Export needed symbol for ZERO_PAGE usage in modules.

From: Aneesh Kumar K.V <aneesh.kumar@linux.vnet.ibm.com>

ext4 use ZERO_PAGE(0) to zero out blocks. We need to export
different symbols in different arch for the usage of ZERO_PAGE
in modules.

Signed-off-by: Aneesh Kumar K.V <aneesh.kumar@linux.vnet.ibm.com>
---

 arch/arm/mm/mmu.c               |    1 +
 arch/m68k/mm/init.c             |    1 +
 arch/powerpc/kernel/ppc_ksyms.c |    2 ++
 arch/sparc/kernel/sparc_ksyms.c |    2 ++
 arch/sparc64/mm/init.c          |    1 +
 5 files changed, 7 insertions(+)


Index: linux-2.6.25-rc6/arch/arm/mm/mmu.c
===================================================================
--- linux-2.6.25-rc6.orig/arch/arm/mm/mmu.c	2008-03-24 14:24:12.000000000 -0700
+++ linux-2.6.25-rc6/arch/arm/mm/mmu.c	2008-03-24 14:33:21.000000000 -0700
@@ -35,6 +35,7 @@ extern pgd_t swapper_pg_dir[PTRS_PER_PGD
  * zero-initialized data and COW.
  */
 struct page *empty_zero_page;
+EXPORT_SYMBOL(empty_zero_page);
 
 /*
  * The pmd table for the upper-most set of pages.
Index: linux-2.6.25-rc6/arch/m68k/mm/init.c
===================================================================
--- linux-2.6.25-rc6.orig/arch/m68k/mm/init.c	2008-03-24 14:24:12.000000000 -0700
+++ linux-2.6.25-rc6/arch/m68k/mm/init.c	2008-03-24 14:33:21.000000000 -0700
@@ -69,6 +69,7 @@ void __init m68k_setup_node(int node)
  */
 
 void *empty_zero_page;
+EXPORT_SYMBOL(empty_zero_page);
 
 void show_mem(void)
 {
Index: linux-2.6.25-rc6/arch/powerpc/kernel/ppc_ksyms.c
===================================================================
--- linux-2.6.25-rc6.orig/arch/powerpc/kernel/ppc_ksyms.c	2008-03-24 14:24:12.000000000 -0700
+++ linux-2.6.25-rc6/arch/powerpc/kernel/ppc_ksyms.c	2008-03-24 14:33:21.000000000 -0700
@@ -192,3 +192,5 @@ EXPORT_SYMBOL(intercept_table);
 EXPORT_SYMBOL(__mtdcr);
 EXPORT_SYMBOL(__mfdcr);
 #endif
+
+EXPORT_SYMBOL(empty_zero_page);
Index: linux-2.6.25-rc6/arch/sparc/kernel/sparc_ksyms.c
===================================================================
--- linux-2.6.25-rc6.orig/arch/sparc/kernel/sparc_ksyms.c	2008-03-24 14:24:12.000000000 -0700
+++ linux-2.6.25-rc6/arch/sparc/kernel/sparc_ksyms.c	2008-03-24 14:33:21.000000000 -0700
@@ -295,3 +295,5 @@ EXPORT_SYMBOL(do_BUG);
 
 /* Sun Power Management Idle Handler */
 EXPORT_SYMBOL(pm_idle);
+
+EXPORT_SYMBOL(empty_zero_page);
Index: linux-2.6.25-rc6/arch/sparc64/mm/init.c
===================================================================
--- linux-2.6.25-rc6.orig/arch/sparc64/mm/init.c	2008-03-24 14:24:12.000000000 -0700
+++ linux-2.6.25-rc6/arch/sparc64/mm/init.c	2008-03-24 14:33:21.000000000 -0700
@@ -159,6 +159,7 @@ extern unsigned int sparc_ramdisk_image;
 extern unsigned int sparc_ramdisk_size;
 
 struct page *mem_map_zero __read_mostly;
+EXPORT_SYMBOL(mem_map_zero);
 
 unsigned int sparc64_highest_unlocked_tlb_ent __read_mostly;
 
